Assets/Tests/InputSystem/Plugins/UITests.cs

```diff
@@ -4016,6 +4016,260 @@ public IEnumerator UI_DisplayIndexMatchesDisplayMultiplePointers()
         Assert.AreEqual(0, scene.parentReceiver.events.Count);
     }
 
+    void ClickButton(Button button)
+    {
+        bool called = false;
+        button.onClick.AddListener(() => { called = true; });
+        button.OnPointerClick(new PointerEventData(EventSystem.current) { button = PointerEventData.InputButton.Left });
+        //Debug.Log(called);
+        Assert.AreEqual(called, true);
+    }
+
+    void ChangeCanvasRenderMode(Canvas canvas, RenderMode mode)
+    {
+        canvas.renderMode = mode;
+    }
+
+    [Test]
+    public void ButtonClickTest()
+    {
+        var allParent = GameObject.Instantiate(Resources.Load("AllParent"));
+
+        var go = GameObject.Find("Left Eye Canvas Screen Camera").transform.Find("Left Button");
+        Button leftButton = go.GetComponent<Button>();
+        Assert.IsNotNull(leftButton, "Can not find Left Button!");
+
+        Button rightButton = GameObject.Find("Right Eye Canvas Screen Camera").transform.Find("Right Button").GetComponent<Button>();
+        Assert.IsNotNull(leftButton, "Can not find Right Button!");
+
+        Canvas leftEyeCanvas = GameObject.Find("Left Eye Canvas Screen Overlay").GetComponent<Canvas>();
+        Assert.IsNotNull(leftEyeCanvas, "Can not find Left Eye Canvas Screen Overlay!");
+
+        Canvas rightEyeCanvas = GameObject.Find("Right Eye Canvas Screen Overlay").GetComponent<Canvas>();
+        Assert.IsNotNull(rightEyeCanvas, "Can not find Right Eye Canvas Screen Overlay!");
+
+        //ClickBothButtonOnCanvasOverlayMode
+        ChangeCanvasRenderMode(leftEyeCanvas, RenderMode.ScreenSpaceOverlay);
+        ChangeCanvasRenderMode(rightEyeCanvas, RenderMode.ScreenSpaceOverlay);
+        ClickButton(leftButton);
+        ClickButton(rightButton);
+
+        //ClickBothButtonOnCanvasCameraMode
+        ChangeCanvasRenderMode(leftEyeCanvas, RenderMode.ScreenSpaceCamera);
+        ChangeCanvasRenderMode(rightEyeCanvas, RenderMode.ScreenSpaceCamera);
+        ClickButton(leftButton);
+        ClickButton(rightButton);
+
+        //ClickBothButtonOnCanvasOverlayMode&CameraMode
+        ChangeCanvasRenderMode(leftEyeCanvas, RenderMode.ScreenSpaceOverlay);
+        ChangeCanvasRenderMode(rightEyeCanvas, RenderMode.ScreenSpaceCamera);
+        ClickButton(leftButton);
+        ClickButton(rightButton);
+
+        GameObject.DestroyImmediate(allParent);
+    }
+
+    [UnityTest]
+    public IEnumerator DragImageTest()
```

Comment:
To be consistent with the conventions of the tests in this file, I think we should break this test into two tests named:


UI_DisplayIndexMatchesDisplayWithMouseDragOnOverlayCanvas
UI_DisplayIndexMatchesDisplayWithMouseDragOnScreenSpaceCanvas


Also, this test is also covered by the same two tests I mentioned previously:
[UI_DisplayIndexMatchesDisplayWithMouseOnOverlayCanvas
UI_DisplayIndexMatchesDisplayWithMouseOnScreenSpaceCanvas](https://github.com/Unity-Technologies/InputSystem/pull/1603/commits/6438dde73bbcb06e997a9019e201ff0c69401f9e#diff-21434067f6fafbece8297b35cc73ac692f2add4300e88703810bf8be4556c91bR3767)

BUT, I think we should go ahead and get this one converted to work with their test framework anyway.